### PR TITLE
Feature 10249 increase postgre sql to min 12 and default 14

### LINKED ISSUE #10249: Increase PostgreSQL minimum version to 12 and default to 14 [3]
### Problem Description
Postgres versions below 11 are no longer supported (https://www.postgresql.org/support/versioning/), but are still mentioned in the server setup instructions for SORMAS.
New versions are released every year and old versions no longer supported after 5 years, so we should also make sure the latest version of postgres is working with SORMAS and is the suggested postgres version when setting up a system.
Besides the security aspect this also blocks us from using new functionality (like the uuid-ossp module that would be very helpful for #7086).

### Proposed Change

Requires #10260.

- [x] Update postgres to version 14 for a sormas instance that was setup with a previous version and check whether it's working
- [x] Setup a new sormas instance with postgres 14 and see whether it's working
- [x] Add a "minimum version 12"-check to server-setup.sh and update the expected default postgres directory to the current version 14
- [x] Update server setup instructions: https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/SERVER_SETUP.md#postgres-database
- [x] Provide a PR for sormas docker to update to postgres 14
- [x] Mention in the [breaking changes section](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/SERVER_UPDATE.md#breaking-updates) that at least Ubuntu LTS 20 (or 18 with postgres 12 from APT) is needed

### Implementation Details

- [ ] Refactor/Update currently used `ExtendedPostgreSQL94Dialect`.

### Additional Information

- This ticket aims to help #6024 which needs Postgres 12

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #10249

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12521

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12521

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12521

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12521

File: sormas-base/setup/server-setup.sh [hunk 1]
```diff
@@ -301,19 +301,32 @@ CREATE EXTENSION IF NOT EXISTS unaccent;
 GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA public TO $DB_USER;
 EOF
 
+PostgreSQL_VERSION = psql --version 2>&1 | sed 's/psql (PostgreSQL) / /;s/\..*$//'
   if [[ ${LINUX} = true ]]; then
     # no host is specified as by default the postgres user has only local access
+    if [ PostgreSQL_VERSION < 12]; then
+        echo "ERROR: Found Database server ${PostgreSQL_VERSION}  This version is too old."
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-base/setup/server-setup.sh]: updated in all places

**Review by MateStrysewske — APPROVED** [review]

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Test failures

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12521

**Comment by sergiupacurariu** [comment]
> Test failures

fixed

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**